lib/types.js

```diff
@@ -178,4 +178,4 @@ export type ChatReadMessage = any
 export type DeviceUsernamePair = {|username: string, devicename: string|}
 /**
  */
-export type NativeStatus = void | {Device?: {name: string}, Username?: string}
+export type UsernameAndDevice = {|username: string, devicename: string|}
```

Comment:
`keybase status -j` can fail, return nothing, return non-json (message to STDOUT), etc. 

What we're doing on `keybase-status.js#L7` is handling the falsy fields to convince Flow that the branch of execution to `return` would not happen unless both fields are truthy. So these properties are very much still optional and should stay that way in the Flow typing. One option is to split this into two different types:

1. `MaybeUsernameAndDevice` for the output from `keybase status -j`
2. `UsernameAndDevice` for the resolved type of the Promise

Reply:
If `keybase status -j` doesn't return a valid username and devicename, won't we just throw an error in which case it doesn't matter what our function returns? We only get `UsernameAndDevice` if we have both a username and device from `keybase status`, right?

Reply:
@nathunsmitty Good point. The previous implementation handled the case of `keybase status -j` succeeding (exit code 0) and the username / device not being present in the output.

As you mentioned, it might be case that the service will always either succeed with valid JSON or error (exit code != 0) and the previous implementation was against a different `keybase status` behavior.

I think you're right about this - I'll take a look at the code and ping someone one CORE to double check. For now this works, thanks for pointing this out!

Reply:
Note to self: **Write a test for this**